### PR TITLE
Homework Podstawy C++ #2

### PR DESCRIPTION
**Pull request by NeoBauer**
Authors: @NeoBauer @mateusz-talma 
NeoBauer added solutions for tasks "AddEven" and "MaxOfVector"
mateusz-talma added solutions for tasks "nwdNww" and "GenereteSequence"

### CODE REVIEW

File: module2/homework/addEven/addEven.hpp [hunk 1]
```diff
@@ -2,7 +2,21 @@
 #include <vector>
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+
+	int sum{};
+
+	for(auto i:numbers)
```
Comment (anraf1001) [inline module2/homework/addEven/addEven.hpp:8]:
```suggestion
	for(const auto& i:numbers)
```

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 2]
```diff
@@ -3,6 +3,15 @@
 #include <limits>
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
+
+int actual_max = vec.at(0);
+	for(auto element:vec)
```
Comment (anraf1001) [inline module2/homework/maxOfVector/maxOfVector.hpp:8]:
```suggestion
	for(const auto& element:vec)
```

File: module2/homework/nwdNww/nwdNww.hpp [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
 }
 
-int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+int NWW(int lhs, int rhs) 
+{
+    int res = 0;
+    if (lhs != 0 && rhs != 0)
+        res = (lhs * rhs) / (NWD(lhs, rhs));
```
Comment (Rosein) [inline module2/homework/nwdNww/nwdNww.hpp:16]: I'm copying piezol's comment from different PR:
There are trivial cases that your code won't work for, ie NWW(1e8, 1e8).
That is due to lhs*rhs overflowing int.
You can fix it by casting lhs to unsigned long long in the multiplication.

File: module2/homework/addEven/addEven.hpp (hunk 1 above)
Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:8]: space between :

File: module2/homework/addEven/addEven.hpp [hunk 4]
```diff
@@ -2,7 +2,21 @@
...
+	{
+
+		if(i%2 == 0 && i != 0) {
+
+			sum+=i;
+
```
Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:14]: to much new line and space between operator like `:` `%` etc...

File: module2/homework/addEven/addEven.hpp [hunk 5]
```diff
@@ -2,7 +2,21 @@
...
+		}	
+
+
+	}
+
+
```
Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:20]: why so much new line?

File: module2/homework/generateSequence/generateSequence.hpp [hunk 6]
```diff
@@ -1,7 +1,21 @@
...
+
+    if (count > 0)
+    {
+        sequence.resize(count);
+        int value = step;
+
+        for(int i = 0; i < sequence.size(); i++)
+        {
+            sequence[i] = value;
+            value += step;
```
Comment (nauka-programowania-MA) [inline module2/homework/generateSequence/generateSequence.hpp:16]: you can use sequence[i] = step * i
or sequence[0] = step; sequence[i] = sequence[i-1] + step;
And you avoid uneccesary variable (value)

File: module2/homework/generateSequence/main.cpp [hunk 7]
```diff
@@ -2,7 +2,7 @@
 #include "generateSequence.hpp"
 
 int main() {
-    for (const int element : generateSequence(10, 5))
+    for (const int element : generateSequence(10,5))
```
Comment (nauka-programowania-MA) [inline module2/homework/generateSequence/main.cpp:5]: SPACE!

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 8]
```diff
@@ -3,6 +3,15 @@
 #include <limits>
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
+
+int actual_max = vec.at(0);
+	for(const auto& element:vec)
```
Comment (nauka-programowania-MA) [inline module2/homework/maxOfVector/maxOfVector.hpp:8]: space

**Comment by ziobron** [comment]
{
  "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635],
  "points": [2, 4, 2, 6, 2, 5, 2, 5],
  "users": ["NeoBauer", "mateusz-talma"]
}

**Comment by ziobron** [comment]
+ points for cooperation

{
"requirement_ids": [1636],
"points": [3],
"users": ["NeoBauer", "mateusz-talma"]
}

**Pull request closed without merging**